teapots/choreographer-30fps/src/main/cpp/CMakeLists.txt

```diff
@@ -17,36 +17,53 @@
 cmake_minimum_required(VERSION 3.4.1)
 
 # build native_app_glue as a static lib
-add_library(native_app_glue STATIC
-    ${ANDROID_NDK}/sources/android/native_app_glue/android_native_app_glue.c)
-
-set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -std=gnu++11 -Wall -fno-exceptions -fno-rtti")
+add_library(nativeAppGlue STATIC
+  ${ANDROID_NDK}/sources/android/native_app_glue/android_native_app_glue.c)
```

Comment:
Also, it helps when crafting projects built for multiple deployment targets to guard any platform specific code or dependencies.  Thus this could be changed to:
if(ANDROID) //ANDROID is defined in the NDK CMake toolchain
include(AndroidNdkModules)
android_ndk_import_module_native_app_glue()
endif()

Now later when you want to use it just add it to the target like so:
if(ANDROID)
target_link_libraries(ChoreographerNativeActivity native_app_glue)
endif(ANDROID)
By breaking up the target_link_library calls you allow platform specific decisions at build time.

Reply:
My first github code review, this comment thread appears collapsed; wanted to ensure this little gem is noticed as it's very useful to be able to avoid hard ndk paths in user land CMakeLists.txt so bump!